libs/colbert/ragstack_colbert/colbert_embedding_model.py

```diff
@@ -53,165 +25,87 @@ class ColbertEmbeddingModel(BaseEmbeddingModel):
-            self.colbert_config.checkpoint,
-            colbert_config=self.colbert_config,
-            verbose=verbose,
-        )
-        self.encoder = CollectionEncoder(
-            config=self.colbert_config, checkpoint=self.checkpoint
+        self._query_maxlen = query_maxlen
+        self._chunk_batch_size = chunk_batch_size
+
+        colbert_config = ColBERTConfig(
+            doc_maxlen=doc_maxlen,
+            nbits=nbits,
+            kmeans_niters=kmeans_niters,
+            nranks=nranks,
+            checkpoint=checkpoint,
         )
-        self.query_tokenizer = QueryTokenizer(self.colbert_config)
-        if self.__cuda:
-            self.checkpoint = self.checkpoint.cuda()
+        self._encoder = TextEncoder(config=colbert_config, verbose=verbose)
 
-    def embed_chunks(
-        self, chunks: List[ChunkData], doc_id: Optional[str] = None
-    ) -> List[EmbeddedChunk]:
+    # implements the Abstract Class Method
+    def embed_texts(self, texts: List[str]) -> List[Embedding]:
         """
-        Encodes a list of text chunks into embeddings, returning them as a list of EmbeddedChunk objects.
-        Each chunk text is converted into a dense vector representation.
+        Embeds a list of texts into their corresponding vector embedding representations.
 
         Parameters:
-            texts (List[str]): The list of chunk texts to be embedded.
-            doc_id (Optional[str]): An optional document identifier. If not provided, a UUID is generated.
+            texts (List[str]): A list of string texts.
 
         Returns:
-            List[EmbeddedChunk]: A list of EmbeddedChunk objects containing the embeddings and document/chunk identifiers.
+            List[Embedding]: A list of embeddings, in the order of the input list
         """
 
-        if doc_id is None:
-            doc_id = str(uuid.uuid4())
-
-        timeout = 60 + len(chunks)
-
-        chunk_data_map: Dict[int, ChunkData] = {}
-        texts: List[str] = []
-
-        for chunk_idx, chunk in enumerate(chunks):
-            chunk_data_map[chunk_idx] = chunk
-            texts.append(chunk.text)
+        chunks = [
+            Chunk(doc_id="dummy", chunk_id=i, text=t) for i, t in enumerate(texts)
+        ]
 
-        embedded_texts = self._encode_texts_using_distributed(texts=texts, timeout=timeout)
+        embedded_chunks = []
 
-        output: List[EmbeddedChunk] = []
-        for embedded_text in embedded_texts:
-            chunk_data = chunk_data_map[embedded_text.original_index]
-            output.append(EmbeddedChunk(
-                data=chunk_data,
-                doc_id=doc_id,
-                chunk_id=embedded_text.chunk_id,
-                embeddings=embedded_text.embeddings,
-            ))
+        for i in range(0, len(chunks), self._chunk_batch_size):
+            chunk_batch = chunks[i : i + self._chunk_batch_size]
+            embedded_chunks.extend(self._encoder.encode_chunks(chunks=chunk_batch))
```

Comment:
encode_chunks calls Colbert doc encoder directly. This will not disperse the load to multiple CUDA. That's why it was run under multiprocessing

Reply:
i tested it with mulitprocessing and it was slower than just using the single encoder.